### PR TITLE
Implemented File ranking system 

### PR DESCRIPTION
**Pull request by Dhruv-Panicker**

### CODE REVIEW

File: algorithim/author_data.py [hunk 1]
```diff
@@ -15,18 +15,20 @@ def check_author_data(author_data, db, ExcelFile, template_author_data):
...
     if author_data1["creator"] == author_data2["creator"]:
-      suspicious_files.setdefault(file_id1, []).append(f"same_creator:{author_data1['creator']}")
-      suspicious_files.setdefault(file_id2, []).append(f"same_creator:{author_data2['creator']}")
+      suspicious_files.setdefault(file_id1, []).append(("author_data", f"same_creator:{author_data1['creator']}", 3))
+      suspicious_files.setdefault(file_id2, []).append(("author_data", f"same_creator:{author_data2['creator']}", 3))
 
     if author_data1["lastModifiedBy"] == author_data2["lastModifiedBy"]:
-      suspicious_files.setdefault(file_id1, []).append(f"same_last_modified_by:{author_data1['lastModifiedBy']}")
-      suspicious_files.setdefault(file_id2, []).append(f"same_last_modified_by:{author_data2['lastModifiedBy']}")
+      suspicious_files.setdefault(file_id1, []).append(("author_data", f"same_last_modified_by:{author_data1['lastModifiedBy']}", 2))
```
[the remark this answers is not in the document]
Comment (Dhruv-Panicker) [reply algorithim/author_data.py]: Format has been resolved

File: algorithim/author_data.py [hunk 2]
```diff
@@ -15,18 +15,20 @@ def check_author_data(author_data, db, ExcelFile, template_author_data):
...
+      suspicious_files.setdefault(file_id2, []).append(("author_data", f"same_last_modified_by:{author_data2['lastModifiedBy']}", 2))
 
     if author_data1["created"] == author_data2["created"]:
-      suspicious_files.setdefault(file_id1, []).append(f"same_creation_date:{author_data1['created']}")
-      suspicious_files.setdefault(file_id2, []).append(f"same_creation_date:{author_data2['created']}")
-
+      auth1_str = author_data1["created"].strftime('%m/%d/%Y')
+      auth2_str = author_data2["created"].strftime('%m/%d/%Y')
+      suspicious_files.setdefault(file_id1, []).append(("author_data", f"same_creation_date:{auth1_str}", 2))
```
[the remark this answers is not in the document]
Comment (Dhruv-Panicker) [reply algorithim/author_data.py]: Format has been resolved

File: algorithim/column_width_data.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    columns2 = file_column_data[file_id2]
+
+    common_unique_columns = set(columns1).intersection(columns2) - template_column_set
+
+    if common_unique_columns:
+      # If there are common columns not in the template, flag both files
+      for file_id in (file_id1, file_id2):
+        if file_id not in suspicious_files:
+          suspicious_files[file_id] = ("column_width", [], 0)
+        suspicious_files[file_id][1].extend(common_unique_columns)
```
[the remark this answers is not in the document]
Comment (Dhruv-Panicker) [reply algorithim/column_width_data.py]: Formatting change to (column_width, [Same column width [ ] as FILE, score)

File: algorithim/column_width_data.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  template_column_set = set(template_column_data)
+
+  # Generate combinations of file names
+  file_combinations = combinations(file_column_data.keys(), 2)
+
+  for file_id1, file_id2 in file_combinations:
+    columns1 = file_column_data[file_id1]
+    columns2 = file_column_data[file_id2]
+
+    common_unique_columns = set(columns1).intersection(columns2) - template_column_set
```
[the remark this answers is not in the document]
Comment (Dhruv-Panicker) [reply algorithim/column_width_data.py]: Column width will now check IDENTICAL oddly sized columns, eliminating all standard column sizes that are a multiple of 5

File: tests/test_scan_results.py [hunk 5]
```diff
@@ -3,23 +3,29 @@
...
-        yield testing_client
+  app.config['TESTING'] = True
+  with app.test_client() as testing_client:
+    with app.app_context():
+      pass
+    yield testing_client
 
-def test_get_fingerprint_results(test_client):
-    with app.app_context(): 
+def test_get_fingerprint_results():
```
[the remark this answers is not in the document]
Comment (Dhruv-Panicker) [reply tests/test_scan_results.py:19]: Fixed fingerprint test format, needs to be verified again with multiple connections

**Review by pJeyakumar — APPROVED** [review]

**Pull request merged**